src/components/trashlookup.jsx

```diff
@@ -1,156 +1,121 @@
+  displayCollectionSechduleText() {
+    const selectedAddress = _.trim(this.state.selectedAddress);
+    return <InformationSection address={selectedAddress} resetForm={this.handleClick} />;
+  }
 
-class Movies extends Component {
-  state = {
-    services: getTrashService(),
-    searchQuery: "",
-    addresses: PostData,
-    issearch: 0,
-    selectedAddress: null
-  };
+  addressData() {
+    let searchQuery = _.trim(this.state.selectedAddress);
+    let filtered = PostData;
+    const isCollectionScheduleTextHidden = this.state.isCollectionScheduleTextHidden;
 
-  // componentDidMount() {
-  // console.log("inside didmount");
-  //const fakeAddress = [{ _id: "", name: "" }, ...getFakeAddress()];
-  // this.setState({ searchQuery: this.ref.searchboxinput.value });
-  // }
-  // componentDidUpdate() {
-  //   console.log("inside componentDidUpdate" + this.state.searchQuery);
-  // }
-  handleAddressSelect = address => {
-    // console.log("address:" + address.address1);
-    this.setState({
-      selectedAddress: address.address1,
-      searchQuery: address.address1,
-      issearch: 3
-    });
-  };
-  handleSearch = query => {
-    console.log("handel Search:" + query);
-    if (query.length === 0) {
-      this.setState({ searchQuery: query, issearch: 0 });
-    } else {
-      this.setState({ searchQuery: query, issearch: 1 });
+    if (searchQuery.length > 0) {
+      if (isCollectionScheduleTextHidden) {
+        filtered = PostData.filter(m => m.address1.toLowerCase().indexOf(searchQuery.toLowerCase()) > -1);
+      } else {
+        filtered = PostData.filter(m => m.address1.toLowerCase() === searchQuery.toLowerCase());
+      }
     }
-  };
+    return { data: filtered };
+  }
+  trashSchedule() {
+    const scheduleType = this.state.scheduleType;
+    //TODO: we will use this later
+    // const dowloadMessage =
+    //   "Download your complete four year scheduled in PDF format";
+    return (
+      <div className="row">
+        <div className="col-5">
+          <ScheduleTable
+            services={getTrashService()}
+            scheduleType={scheduleType}
+            renderWeekOfDay={this.renderWeekOfDay}
+          />
+          <h6>Download</h6>
+          {/* <a href="/test">{dowloadMessage}</a> */}
+        </div>
+      </div>
+    );
+  }
 
-  handleSearchClick = query => {
-    if (this.state.searchQuery.length === 0) {
-      alert("search criteria misssing");
-    } else {
-      this.setState({ issearch: 2 });
-    }
-    console.log("button is click");
+  renderWeekOfDay = type => {
+    const { scheduleType } = this.state;
+    const trash = scheduleType.trash.toLowerCase();
+    const recycle = scheduleType.recycle.toLowerCase();
+    return type === trash
+      ? this.trashNextCollectionDate()
+      : type === recycle
+      ? this.recylceNextCollectionDate()
+      : this.leafNextCollectionDate();
   };
+  leafNextCollectionDate() {
+    const { dateFormat, dayOfWeek } = this.state;
```

Comment:
it looks like some of this logic is shared between the different colleciton date function, you should try to combine the shared logic into one method, so that you aren't repeating your self. Doing so, makes the code harder to maintain